plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/connection/ConnectionsRegistry.java

```diff
@@ -150,54 +142,37 @@ public boolean remove(IConnection connection) {
 		}
 	}
 
-	private void fireChange(IConnection  connection, int event) {
+	private void fireChange(IConnection connection, int event) {
 		fireChange(connection, event, null, null, null);
 	}
-	private void fireChange(IConnection  connection, int event, String property, Object oldValue, Object newValue) {
+
+	private void fireChange(IConnection connection, int event, String property, Object oldValue, Object newValue) {
 		if (connection == null) {
 			return;
 		}
-		Job job = new AbstractDelegatingMonitorJob("Connection Changed") {
-		
-			@Override
-			protected IStatus doRun(IProgressMonitor monitor) {
-				try {
-					Iterator<IConnectionsRegistryListener> i = getListeners().iterator();
-					while (i.hasNext()) {
-						IConnectionsRegistryListener l = i.next();
-						switch (event) {
-						case ADDED:
-							l.connectionAdded(connection);
-							break;
-						case REMOVED:
-							l.connectionRemoved(connection);
-							break;
-						case CHANGED:
-							l.connectionChanged(connection, property, oldValue, newValue);
-							break;
-							
-						default:
-							break;
-						}
-					}
-				}catch(Exception e) {
-					return new Status(IStatus.ERROR,OpenShiftCommonCoreActivator.PLUGIN_ID,"Exception while notifying of a connection change",e);
-				}	
-				return Status.OK_STATUS;
+
```

Comment:
This is actually an essential change. Putting the notification in a Job will potentially reorder changes (jobs run in parallel). For example, I had the case where an update notification was arriving after a delete notification (even though the delete happened last). If a client takes a long time reacting to a change notification it is up to the client to move the work to a background thread. Do you have a particular scenario in mind?